joern/joernsteps/cfg.groovy

```diff
@@ -8,3 +8,24 @@ Gremlin.defineStep('toExitNode', [Vertex,Pipe], {
 	_().transform{ queryNodeIndex('functionId:' + it.functionId + " AND type:CFGExitNode ") }
 	.scatter()
 })
+
+/**
+   Search the CFG breadth-first so that we can keep track of all nodes we've visited in
+    the entire search rather than just along the current path (massive optimization for
+    high branching-factor CFGs, e.g. state machines).
+*/
+Object.metaClass._reachableCfgNodes = { curNodes, visited ->
+  nextNodes = curNodes._().out('FLOWS_TO').dedup.toSet() - visited
```

Comment:
Just noticed this dedup is now redundant, I was using lists before and it was necessary then.